osquery/tables/networking/windows/win_sockets.h

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <ws2tcpip.h>
+#define WIN32_LEAN_AND_MEAN
+#include <Iphlpapi.h>
+#include <windows.h>
+#include <winsock2.h>
+
+namespace osquery {
+namespace tables {
+
+enum class WinSockTableType { tcp, tcp6, udp, udp6 };
+
+class WinSockets {
```

Comment:
Since this class has raw pointers it's best to include:
```cpp
#include <boost/noncopyable.hpp>
```
above.

Then having this class inherit:
```cpp
class WinSockets : private boost::noncopyable {
```